drivers/input/input_stmpe811.c

```diff
@@ -0,0 +1,469 @@
+/**
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT st_stmpe811
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/input/input.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(stmpe811, CONFIG_INPUT_LOG_LEVEL);
+
+#include "input_stmpe811.h"
+
+struct stmpe811_config {
+	struct i2c_dt_spec bus;
+	struct gpio_dt_spec int_gpio;
+	uint16_t screen_width;
+	uint16_t screen_height;
+	uint8_t panel_driver_settling_time;
+	uint8_t touch_detect_delay;
+	uint8_t touch_average_control;
+	uint16_t hold_timeout;
+	uint16_t polling_period;
+	uint16_t release_counter_threshold;
+};
+
+struct stmpe811_data {
+	const struct device *dev;
+	struct k_work work;
+	struct k_work_delayable polling_work;
+	struct gpio_callback int_gpio_cb;
+	uint32_t poll_timeout_us;
+	uint32_t poll_period_us;
+	uint32_t poll_release_threshold;
```

Comment:
these should go, see my comment in the init function

Reply:
Done